src/components/ReadingGoalPage/ReadingGoalWeekPreviewTab.tsx

```diff
@@ -86,44 +77,54 @@ const ReadingGoalWeekPreviewTab: React.FC<ReadingGoalTabProps> = ({ state, nav }
       );
     }
 
-    if (data.data.type === ReadingGoalType.TIME) {
-      return `${t('reciter:read')} ${secondsToReadableFormat(data.data.dailyAmount, t, lang)}`;
+    const numberAmount = day.amount as number;
+    if (type === ReadingGoalType.TIME) {
+      return `${t('reciter:read')} ${secondsToReadableFormat(numberAmount, t, lang)}`;
     }
 
-    const pages = Number(data.data.dailyAmount.toFixed(2));
+    const pages = Number(numberAmount.toFixed(2));
     return `${t('reciter:read')} ${t('x-pages', {
       count: pages,
       pages: toLocalizedNumber(pages, lang),
     })}`;
   };
 
+  const getSkeleton = () => {
+    return Array.from({
+      length: Math.min(state.period === ReadingGoalPeriod.Continuous ? state.duration : 7, 7),
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+    }).map((_, idx) => (
+      // eslint-disable-next-line react/no-array-index-key
+      <li key={idx} className={styles.dayPreview}>
+        <Spinner />
+      </li>
+    ));
+  };
+
   return (
     <>
       <div className={styles.titleContainer}>
         <h1 className={styles.title}>{t('preview-schedule.title')}</h1>
         <p className={styles.subtitle}>{t('preview-schedule.description')}</p>
       </div>
       <ol className={classNames(styles.optionsContainer, styles.previewWrapper)}>
-        {/* eslint-disable-next-line @typescript-eslint/naming-convention */}
-        {(data?.data && 'ranges' in data.data ? data.data.ranges : week).map((_, idx: number) => {
-          const day = week[idx];
-
-          return (
-            <li key={day.getDate()} className={styles.dayPreview}>
-              <HoverablePopover content={dateToReadableFormat(day, lang)}>
-                <h3>{t('day-x', { day: toLocalizedNumber(idx + 1, lang) })}</h3>
-              </HoverablePopover>
-
-              {isLoading ? (
-                <div>
-                  <Spinner />
-                </div>
-              ) : (
-                <p>{getDailyAmount(idx)}</p>
-              )}
-            </li>
-          );
-        })}
+        {isValidating
+          ? getSkeleton()
+          : data.data.week.map((day: EstimatedReadingGoal['week'][number], idx: number) => {
```

Comment:
Re: `EstimatedReadingGoal['week'][number]`: We usually add the response from BE as a separate type and automatically infer when using it instead of defining it on usage. E.g. `types/auth/UserPreferencesResponse.ts`.